### PR TITLE
Add Gemm instances for performance improvement

### PR DESCRIPTION
**Pull request by zjing14**

### CODE REVIEW

File: include/ck/utility/type_convert.hpp [hunk 1]
```diff
@@ -140,6 +140,36 @@ inline __host__ __device__ float type_convert<float, f8_t>(f8_t x)
...
+}
+
+template <>
+inline __host__ __device__ half2_t type_convert<half2_t, float2_t>(float2_t x)
+{
+
+    const vector_type<float, 2> f32x2_v(x);
+    const auto y = __builtin_amdgcn_cvt_pkrtz(f32x2_v.template AsType<float>()[Number<0>{}],
+                                              f32x2_v.template AsType<float>()[Number<1>{}]);
+    return bit_cast<half2_t>(y);
```
Comment (geyyer) [inline include/ck/utility/type_convert.hpp:170]: Do we need to bit_cast here or y is already a half2_t?

File: include/ck/utility/type_convert.hpp [hunk 2]
```diff
@@ -140,6 +140,36 @@ inline __host__ __device__ float type_convert<float, f8_t>(f8_t x)
...
+    constexpr bool negative_zero_nan = true;
+    const auto f8x2_v                = vector_type<f8_t, 2>(x);
+    vector_type<float, 2> f32x2_v;
+    f32x2_v.template AsType<float>()(Number<0>{}) =
+        utils::cast_from_f8<f8_t, float, negative_zero_nan>(
+            f8x2_v.template AsType<f8_t>()[Number<0>{}]);
+    f32x2_v.template AsType<float>()(Number<1>{}) =
+        utils::cast_from_f8<f8_t, float, negative_zero_nan>(
+            f8x2_v.template AsType<f8_t>()[Number<1>{}]);
+    return f32x2_v.template AsType<float2_t>()[Number<0>{}];
```
Comment (geyyer) [inline include/ck/utility/type_convert.hpp:159]: Does index 0 mean that we return only the 0th element of a vector?

**Comment by illsilin** [comment]
Looks like one of the tests is still failing in CI:

/home/jenkins/workspace/_kernel_gemm_splitk_improve_kpad/test/grouped_gemm/test_grouped_gemm_interface.cpp:112: Failure
Value of: DefaultGGemmInstance{}.IsSupported(Ms, Ns, Ks, StrideAs, StrideBs, StrideCs, kbatch)
  Actual: false
Expected: true
[  FAILED  ] TestGGemmSplitKInterface_MKNKMN.KLoops (210 ms)

http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2Fcomposable_kernel/detail/gemm_splitk_improve_kpad/12/pipeline

**Review by geyyer — APPROVED** [review]
LGTM!

File: include/ck/utility/type_convert.hpp (hunk 2 above)
Comment (zjing14) [reply include/ck/utility/type_convert.hpp:159]: Yes. There is only 1 element in the vector of float2_t.

File: include/ck/utility/type_convert.hpp (hunk 1 above)
Comment (zjing14) [reply include/ck/utility/type_convert.hpp:170]: The return type is already half2_t, but for some reason is not compatible with ck::half2_t.

**Pull request merged**